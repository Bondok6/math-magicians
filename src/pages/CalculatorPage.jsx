import React from 'react';
import Calculator from '../components/Calculator';
import '../styles/CalculatorPage.css';

const CalculatorPage = () => (
  <section className=" container calculator-section">
    <h2> Let&#39;s do some math ! </h2>
    <div className="calculator">
      <Calculator />
    </div>
  </section>
);

export default CalculatorPage;
