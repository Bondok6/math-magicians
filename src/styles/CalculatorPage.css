.calculator-section {
  width: 100vw;
  min-height: 80vh;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.calculator-section h2 {
  font-size: 4rem;
  align-self: flex-start;
  margin-top: 10rem;
}

.calculator {
  width: 60%;
}

@media screen and (max-width: 800px) {
  .calculator-section {
    flex-direction: column;
  }

  .calculator {
    width: 100%;
  }
}
