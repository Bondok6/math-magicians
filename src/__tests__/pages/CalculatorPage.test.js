import { render } from '@testing-library/react';
import CalculatorPage from '../../pages/CalculatorPage';

it('test calculator component', async () => {
  const tree = render(<CalculatorPage />);
  expect(tree).toMatchSnapshot();
});
