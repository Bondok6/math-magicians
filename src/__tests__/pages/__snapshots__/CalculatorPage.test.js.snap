// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test calculator component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class=" container calculator-section"
      >
        <h2>
           Let's do some math ! 
        </h2>
        <div
          class="calculator"
        >
          <div
            class="m-3 row"
          >
            <div
              class="col"
            >
              <div
                class="w-100 p-2 display"
                data-testid="display"
              />
            </div>
          </div>
          <div
            class="m-3 row"
          >
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                AC
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                +/-
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                %
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-warning"
                type="button"
              >
                รท
              </button>
            </div>
          </div>
          <div
            class="m-3 row"
          >
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                7
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                8
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                9
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-warning"
                type="button"
              >
                x
              </button>
            </div>
          </div>
          <div
            class="m-3 row"
          >
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                4
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                5
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                6
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-warning"
                type="button"
              >
                -
              </button>
            </div>
          </div>
          <div
            class="m-3 row"
          >
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                1
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                2
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                3
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-warning"
                type="button"
              >
                +
              </button>
            </div>
          </div>
          <div
            class="m-3 row"
          >
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                0
              </button>
            </div>
            <div
              class="col-6"
            >
              <button
                class="w-100 p-4 btn btn-primary"
                type="button"
              >
                .
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="w-100 p-4 btn btn-warning"
                type="button"
              >
                =
              </button>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class=" container calculator-section"
    >
      <h2>
         Let's do some math ! 
      </h2>
      <div
        class="calculator"
      >
        <div
          class="m-3 row"
        >
          <div
            class="col"
          >
            <div
              class="w-100 p-2 display"
              data-testid="display"
            />
          </div>
        </div>
        <div
          class="m-3 row"
        >
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              AC
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              +/-
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              %
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-warning"
              type="button"
            >
              รท
            </button>
          </div>
        </div>
        <div
          class="m-3 row"
        >
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              7
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              8
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              9
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-warning"
              type="button"
            >
              x
            </button>
          </div>
        </div>
        <div
          class="m-3 row"
        >
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              4
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              5
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              6
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-warning"
              type="button"
            >
              -
            </button>
          </div>
        </div>
        <div
          class="m-3 row"
        >
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              1
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              2
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              3
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-warning"
              type="button"
            >
              +
            </button>
          </div>
        </div>
        <div
          class="m-3 row"
        >
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              0
            </button>
          </div>
          <div
            class="col-6"
          >
            <button
              class="w-100 p-4 btn btn-primary"
              type="button"
            >
              .
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="w-100 p-4 btn btn-warning"
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
